nvm.sh

```diff
@@ -335,21 +335,40 @@ nvm_find_nvmrc() {
   fi
 }
 
-# Obtain nvm version from rc file
+# Obtain the nvm version from the .nvmrc file
 nvm_rc_version() {
+  # Check if quiet mode is on
+  local quiet="false"
+  if [ "$1" == "--silent" ]; then
+    quiet="true"
+  fi
```

Comment:
```suggestion
  local NVM_USE_SILENT
  NVM_USE_SILENT=false
  if [ "${1-}" = '--silent' ]; then
    NVM_USE_SILENT=true
  fi
```
due to ksh, all variables must be declared on one line with `local`, and then assigned on a separate line.

additionally, variable names should be prefixed with NVM_ so that in ksh, they won't clobber user shell vars.
(also, `==` only works in bash/zsh)